core/src/test/java/io/grpc/ContextTest.java

```diff
@@ -407,4 +644,61 @@ public void innerDeadlineCompletesBeforeOuter() throws Exception {
     assertTrue(base.cause() instanceof TimeoutException);
     assertNotSame(base.cause(), child.cause());
   }
+
+  @Test
+  public void cancellationCancelsScheduledTask() {
+    ScheduledExecutorService service = SharedResourceHolder.get(Context.SCHEDULER);
+    try {
+      ThreadPoolExecutor executor = (ThreadPoolExecutor) service;
+      //executor.setRemoveOnCancelPolicy(true);
+      executor.purge();
+      assertEquals(0, executor.getQueue().size());
+      Context.CancellableContext base = Context.current().withDeadlineAfter(1, TimeUnit.DAYS);
+      assertEquals(1, executor.getQueue().size());
+      base.cancel(null);
+      executor.purge();
+      assertEquals(0, executor.getQueue().size());
+    } finally {
+      SharedResourceHolder.release(Context.SCHEDULER, service);
+    }
+  }
+
+  @Test
+  public void testScheduler() throws Exception {
+    assertEquals("context-scheduler", Context.SCHEDULER.toString());
+    ScheduledExecutorService service = Context.SCHEDULER.create();
+    try {
+      assertFalse(service.isShutdown());
+      final AtomicReference<String> threadName = new AtomicReference<String>();
+      final AtomicReference<Boolean> threadIsDaemon = new AtomicReference<Boolean>();
+      final CountDownLatch ran = new CountDownLatch(1);
+      service.execute(new Runnable() {
+        @Override
+        public void run() {
+          threadName.set(Thread.currentThread().getName());
+          threadIsDaemon.set(Thread.currentThread().isDaemon());
+          ran.countDown();
+        }
+      });
+      ran.await();
```

Comment:
any reason to not just get the future from service.execute and call .get() on that?